### ISSUE TITLE
counterfactuals

### ISSUE THREAD

**Issue by bblfish** [BartoszMilewski/Publications#8]
In chapter 1 you mention counterfactuals as equivalent to 0 the initial object. 
But they have a lot more structure than 0.
An example taken from the 1973 book
[Counterfactuals](https://www.wiley.com/en-us/Counterfactuals-p-9780631224259).

is "If kangaroos had no tails they would topple over". It could be true that they have no tails in which case it is a normal if then conditional, if the antecedent is not true then one has to look at the closes possibilities. 

Your "If wishes were horses, then ..." is also a counterfactual, but it has a necessarily false antecedent, and so I think the whole sentence is true.

**Comment by BartoszMilewski** [comment]
I think it's just the imprecise nature of human language. We have a lot of additional context in mind, and we can imagine tailless kangaroos. In a Universe where kangaroos are tailless, it's possible that they would have other ways of balancing themselves. And if we consider that a kangaroo may be born with a defect, or lose its tail in an accident, then this is not a counterfactual.

**Comment by bblfish** [comment]
Yes, counterfactuals logic topologises context: it brings in a notion of closeness between worlds, and what is close depends on where you are, or what you think is close depends on where you believe you are. 
I wrote about that in a Zulip [chat on psychology](https://mattecapu.github.io/ct-zulip-archive/stream/229111-general/topic/category.20theory.20in.20human.20psychology.html#232317132) which can be read with this image in mind (which I uploaded [a little later](https://mattecapu.github.io/ct-zulip-archive/stream/229111-general/topic/category.20theory.20in.20human.20psychology.html#232711417)   in the discussion ).
![image](https://user-images.githubusercontent.com/124506/125176740-31d18e00-e1d6-11eb-9ec6-8d506d34ba56.png)

**Comment by bblfish** [comment]
Just came back to reading your book, updated a lot of course, and so I stumbled upon the same sentence and remembered this conversation. 

> I think it's just the imprecise nature of human language. We have a lot of additional context in mind, and we can imagine tailless kangaroos. In a Universe where kangaroos are tailless, it's possible that they would have other ways of balancing themselves. And if we consider that a kangaroo may be born with a defect, or lose its tail in an accident, then this is not a counterfactual.

yes, you are reasoning correctly. Whether or not it is true that "if kangaroos have no tails, they would topple over" depends on where you think the actual world is and which the other closest ones are. 

I have found videos of tailless kangaroos that don't topple over, and that should form a proof that the counterfactual is false, as the closest world to the actual world is the actual world, and this shows one kangaroo that does not topple over. 

Had we been in a different world (it could be that the video I saw was a deep fake, for some bizaare reason) the truth value of the counterfactual could have been less clear.  That is, without the video, our location in the space of possibilities is much vaguer. But the reasoning could have been the following. The closest possible worlds are those that have the same physical laws, and where we have the least change from actuallity. So we could look at the genetic code of the kangaroo and see if any minor changes would lead to tailless kangaroos. By considering all those likely changes and all the resulting kangaroos, one could come to a conclusion as to whether or not the counterfactual is true.

The point is that counterfactuals can themselves be true or false. This happens when one adds the extra structure of a distance relation - or metric - on possible worlds.  So my guess is that counterfactuals require enriched categories to really make an interesting appearance. Without those we are left stunted counterfactuals such as  "If wishes were horses then beggars would ride" . 

It turns out counterfactuals are helpful in analysing causality and knowledge btw.
 * [Causation and Counterfactuals](https://mitpress.mit.edu/9780262532563/causation-and-counterfactuals/) a 2004 book that brings together some of the debates on the subject
 * [Causality: Models Reasoning and Inference](http://bayes.cs.ucla.edu/BOOK-2K/) by Judea Pearl - is a lot more statistically based. I have not read it but he is in the tradition of thinking of causality counterfactually.... It would be interesting to have a Category Theory view of what is going on there.
 * Philosophical Explanations by Nozick, gave a very intersting analysis of knowledge in terms of counterfactuals
